utils.go

```diff
@@ -230,11 +230,12 @@ func DefaultStoreOptions(rootless bool, rootlessUID int) (StoreOptions, error) {
 			}
 			if storageOpts.GraphRoot == "" {
 				storageOpts.GraphRoot = defaultRootlessGraphRoot
-			} else if storageOpts.RootlessStoragePath != "" {
+			}
+			if storageOpts.RootlessStoragePath != "" {
 				splitPaths := strings.SplitAfter(storageOpts.RootlessStoragePath, "$")
-				validEnv := regexp.MustCompile(`^(HOME|USER|UID)[^a-zA-Z]`).MatchString
+				validEnv := regexp.MustCompile(`^(HOME|USER|UID)([^a-zA-Z]|$)`).MatchString
```

Comment:
Why do we need to change the regex?

@QiWang19, could you break the code in this if branch into a dedicated function and add unit tests? We have a long tradition in regressing with parsing code.